test/e2e/e2e_visibility_on_demand_test.go

```diff
@@ -671,7 +674,368 @@ var _ = Describe("VisibilityOnDemand", Label("visibility-on-demand"), Ordered, f
 				"X-Kubernetes-Pf-Flowschema-Uid header does not match \"visibility\" FlowSchema UID for LocalQueue")
 
 		})
+	})
+
+	When("PendingWorkloads list should be checked for LWS workloads", func() {
+		var (
+			labelKey   = testutils.OpenShiftManagedLabel
+			labelValue = trueLabelValue
+		)
+
+		It("Should show pending LWS workloads ordered by priority and admit them sequentially based on resource availability", func(ctx context.Context) {
+			var (
+				clusterPendingWorkloads *visibilityv1beta1.PendingWorkloadsSummary
```

Comment:
I created https://github.com/openshift/kueue-operator/issues/1101 as a tech debt item we need to clean up.

For Kueue 0.15 and on, Kueue now has v1beta2 APIs. It is probably best to start using the v1beta2 apis whever we can.